Game/Scripts/RangedFastAttackBehaviourScript.cpp

```diff
@@ -73,36 +141,82 @@ void RangedFastAttackBehaviourScript::PerformAttack()
 	script->GetScript()->SetOwner(bullet);
 
 	bullet->GetComponent<RangedFastAttackBullet>()->SetBulletVelocity(bulletVelocity);
+	bullet->GetComponent<RangedFastAttackBullet>()->SetBulletDamage(AttackDamage);
 
 	// Once the engine automatically runs the Start() for newly created objects, delete this line
 	script->Start();
 
-	lastAttackTime = SDL_GetTicks() / 1000.0f;
 	audioSource->PostEvent(AUDIO::SFX::NPC::DRON::SHOT_01);
 
+	currentConsecutiveShots += 1.0f;
+	if (currentConsecutiveShots < numConsecutiveShots)
+	{
+		isConsecutiveShooting = true;
+		nextShotDuration = minTimeConsecutiveShot + static_cast<float>(std::rand()) / (static_cast <float>(RAND_MAX /(maxTimeConsecutiveShot - minTimeConsecutiveShot)));
```

Comment:
Do we need the bigger cast to float?
```suggestion
		nextShotDuration = minTimeConsecutiveShot + std::rand() / (static_cast <float>(RAND_MAX /(maxTimeConsecutiveShot - minTimeConsecutiveShot)));
```
Also, maybe it's just me, but I'm not sure what exactly we're doing here. Looks like we want a random number in the range [minTimeConsecutiveShot, maxTimeConsecutiveShot]? If that's the case, this looks excessively complex, why not just
```suggestion
		nextShotDuration = minTimeConsecutiveShot + std::rand() % (maxTimeConsecutiveShot - minTimeConsecutiveShot + 1);
```